graspologic/preprocessing/graph_cuts.py

```diff
@@ -0,0 +1,383 @@
+        exclusive or inclusive. Allowed values are
+        - ``LARGER_THAN_INCLUSIVE``
+        - ``LARGER_THAN_EXCLUSIVE``
+        - ``SMALLER_THAN_INCLUSIVE``
+        - ``SMALLER_THAN_EXCLUSIVE``
+    weight_attribute : str
+        The weight attribute name in the edge's data dictionary. Default is `weight`.
+    prune_isolates : bool
+        If true, remove any vertex that no longer has an edge.  Note that this only prunes vertices which have edges to
+        be pruned; any isolate vertex prior to any edge cut will be retained.
+
+    Returns
+    -------
+    nx.Graph
+        Pruned copy of the graph
+
+    Notes
+    -----
+    Edges without a `weight_attribute` field will be excluded from these cuts.  Enable logging to view any
+        messages about edges without weights.
+    """
+    filter_by = _filter_function_for_make_cuts(cut_threshold, cut_process)
+    if not isinstance(cut_threshold, int) and not isinstance(cut_threshold, float):
+        raise TypeError(
+            f"cut_threshold must be of type int or float; you provided: {type(cut_threshold)}"
+        )
+
+    logger = logging.getLogger(__name__)
+    graph_copy = graph.copy()
+    edge_weights: List[Tuple[Tuple[Any, Any], Union[int, float, None]]] = [
+        ((source, target), weight)
+        for source, target, weight in graph.edges(data=weight_attribute)
+    ]
+    none_weights: List[Tuple[Tuple[Any, Any], None]] = [
+        (edge, weight) for edge, weight in edge_weights if weight is None
+    ]
+    actual_weights: List[Tuple[Tuple[Any, Any], Union[int, float]]] = [
+        (edge, weight) for edge, weight in edge_weights if weight is not None
+    ]
+
+    if len(none_weights) != 0:
+        logger.warning(
+            f"Graph contains {len(none_weights)} edges with no {weight_attribute}."
+            + f"Ignoring these when cutting by weight"
+        )
+
+    edges_to_cut = [x for x in actual_weights if filter_by(x)]
+    for edge, weight in edges_to_cut:
+        source, target = edge
+        if (
+            source in graph_copy
+            and target in graph_copy
+            and target in graph_copy[source]
+        ):
+            graph_copy.remove_edge(source, target)
+        if prune_isolates:
+            if len(graph_copy[source]) == 0:
+                graph_copy.remove_node(source)
+            if len(graph_copy[target]) == 0:
+                graph_copy.remove_node(target)
+
+    return graph_copy
+
+
+def histogram_degree_centrality(
+    graph: nx.Graph,
+    bin_directive: Union[int, List[Union[float, int]], np.ndarray, str] = 10,
+) -> DefinedHistogram:
+    """
+    Generates a histogram of the vertex degree centrality of the provided graph. Histogram function is fundamentally
+        proxied through to numpy's `histogram` function, and bin selection follows `numpy.histogram` processes.
+
+    Parameters
+    ----------
+
+    graph : nx.Graph
```

Comment:
do these functions in this PR work on DiGraph? multigraph?

Reply:
That's a good question - my guess is the edge cuts in specific get... squirrelly?  on multigraph.  Is the edge cut one of the connections between two nodes or is it all of them if they don't sum up enough?  I think in the short term I'm going to stick with nx.Graph only, and then after I've proved/disproved it works with others, I'll relax the requirements to Graph vs. DiGraph.  

My best is multigraph varietals is just going to be out of luck entirely, and I'm kinda okay with that.